store/store.go

```diff
@@ -85,6 +85,13 @@ var defaultRetryStrategy = retry.LimitCount(5, retry.LimitTime(38*time.Second,
 	},
 ))
 
+var connCheckStrategy = retry.LimitCount(3, retry.LimitTime(38*time.Second,
```

Comment:
Thanks for this!

Reply:
I don't think we need to retry here, or at least we should not treat HTTP errors as failure, only a failure to connect to the host.